test/e2e/sched/sched_test.go

```diff
@@ -192,5 +197,88 @@ var _ = Describe("[Scheduler] imageReplacement", func() {
 				return true
 			}).WithTimeout(time.Minute * 2).WithPolling(time.Second * 30).Should(BeTrue())
 		})
+		It("should reflect changes in cacheResyncPeriod when configured", func() {
+			deployment, err := podlist.With(e2eclient.Client).DeploymentByOwnerReference(context.TODO(), nroSchedObj.UID)
+			Expect(err).ToNot(HaveOccurred())
+			podList, err := podlist.With(e2eclient.Client).ByDeployment(context.TODO(), *deployment)
+			Expect(err).NotTo(HaveOccurred())
+			Expect(podList).ToNot(BeEmpty(), "cannot find any pods for DP %s/%s", deployment.Namespace, deployment.Name)
+			uid := &podList[0].UID
+
+			var t time.Duration
+			if nroSchedObj.Spec.CacheResyncPeriod != nil {
+				// change to something different from the current spec
+				t = nroSchedObj.Spec.CacheResyncPeriod.Duration * 2
+			} else {
+				t = 5 * time.Minute
+			}
+			nroSchedObj.Spec.CacheResyncPeriod = &metav1.Duration{Duration: t}
+			Expect(e2eclient.Client.Update(context.TODO(), nroSchedObj)).ToNot(HaveOccurred())
+
+			By("checking cacheResyncPeriod under the CR's Status")
+			Eventually(func() bool {
+				ExpectWithOffset(1, e2eclient.Client.Get(context.TODO(), client.ObjectKeyFromObject(nroSchedObj), nroSchedObj)).ToNot(HaveOccurred())
+				return nroSchedObj.Spec.CacheResyncPeriod.Duration == nroSchedObj.Status.CacheResyncPeriod.Duration
+			}).WithTimeout(time.Minute*2).WithPolling(time.Second*10).Should(BeTrue(), "cacheResyncPeriod not updated under the status; want: %d, got %d",
```

Comment:
OTOH this `Eventually` block should be distinct from the one suggested above (= it seems fine as it is)

Reply:
If failing to `Get` is a possible option then we should drop the `Expect` shouldn't we?

Reply:
possibly yes. Not blocking.